src/vs/workbench/api/common/extHostFileSystemEventService.ts

```diff
@@ -133,6 +133,38 @@ interface IExtensionListener<E> {
 	(e: E): any;
 }
 
+class LazyRevivedFileSystemEvents implements FileSystemEvents {
+
+	constructor(private readonly _events: FileSystemEvents) { }
+
+	private _created: URI[] | undefined = undefined;
+	get created(): URI[] {
+		if (!this._created) {
+			this._created = this._events.created.map(created => URI.revive(created));
```

Comment:
Save a function call with `.map(URI.revive)`?

Reply:
Yes I had that initially but TypeScript complains about `URI[]` potentially being `(URI | undefined)[]`. I changed it but it needs a cast still.